hadoop-ozone/ozone-manager/src/main/java/org/apache/hadoop/ozone/om/OmMetadataManagerImpl.java

```diff
@@ -1293,39 +1294,69 @@ public List<SnapshotInfo> listSnapshot(String volumeName, String bucketName)
           BUCKET_NOT_FOUND);
     }
 
-    String prefix = getBucketKey(volumeName, bucketName + OM_KEY_PREFIX);
+    String prefix;
+    if (StringUtil.isNotBlank(snapshotPrefix)) {
+      prefix = getOzoneKey(volumeName, bucketName, snapshotPrefix);
+    } else {
+      prefix = getBucketKey(volumeName, bucketName + OM_KEY_PREFIX);
+    }
+
+    String seek;
+    if (StringUtil.isNotBlank(prevSnapshot)) {
+      // Seek to the specified snapshot.
+      seek = getOzoneKey(volumeName, bucketName, prevSnapshot);
```

Comment:
We can probably append the '\0' (char with ascii value 0 to ignore the previous value. So we wouldn't really have to check the previous it would be ignored automatically

Reply:
Something like if previous snapshot is snap1 then we can append '\0' to snap1 to make it 'snap1\0' which would be the next string in lexicographical ordering.